src/editors/application/analysis-features/enabled-analysis-features-repository.php

```diff
@@ -44,12 +36,32 @@ public function __construct( Analysis_Feature_Interface ...$plugin_features ) {
 	 * @return Analysis_Features_List The analysis list.
 	 */
 	public function get_enabled_features(): Analysis_Features_List {
-		if ( \count( $this->enabled_analysis_features->parse_to_legacy_array() ) === 0 ) {
-			foreach ( $this->plugin_features as $plugin_feature ) {
+		$enabled_analysis_features = new Analysis_Features_List();
+		foreach ( $this->plugin_features as $plugin_feature ) {
+			$analysis_feature = new Analysis_Feature( $plugin_feature->is_enabled(), $plugin_feature->get_name(), $plugin_feature->get_legacy_key() );
+			$enabled_analysis_features->add_feature( $analysis_feature );
+		}
+
+		return $enabled_analysis_features;
+	}
+
+	/**
+	 * Returns the analysis list for the given names.
+	 *
+	 * @param array<string> $feature_names The feature names to include.
+	 *
+	 * @return Analysis_Features_List The analysis list.
+	 */
+	public function get_enabled_features_by_keys( array $feature_names ): Analysis_Features_List {
```

Comment:
This shouldn't be `get_enabled_features_by_keys()` but rather `get_features_by_keys()` because it doesn't return enabled features but all features.

But that would require cascading that change to the `get_enabled_features()` function and also the class name etc. so let's not do that here, just something to mention to get it off my chest 😅

Reply:
Solved!